include/boost/url/bnf/parse.hpp

```diff
@@ -16,58 +16,77 @@
+
+   A BNF element:
+
+    - Is default constructible
+    - Holds the result, usually as a reference, of parsing
+    - Has a nested value_type
+    - Has `value_type& operator*() const noexcept`
+
+   Users should define a free function `parse` as a customization point defining how to parse their
+   BNF elements as part of the same architecture that might include arbitrary BNF elements in expressions.
+
+   These new function overloads may be defined in other namespaces. As with `std::swap`, the design relies
+   on Argument Dependent Lookup (ADL) to find these overloads.
+
+*/
+
+
 namespace boost {
 namespace urls {
 namespace bnf {
 
-/*  Design
-
-    - An element is default constructible
-    - Holds the result of parsing
-    - overloads the free function `parse`
-    - has a nested value_type
-    - has `value_type& operator*() const noexcept`
-*/
 
-/** @brief Parse a literal character
+/** Parse a substring as a BNF literal character
 
    This function parses the string defined by the first chars in the
-   range of chars [@ref it, @ref end) as a literal character.
+   range of chars [`it`, `end`) as a literal character.
 
-   The function returns `true` and resets @ref ec when the input is
-   parsed successfully. The initial character in the range is immediately
-   consumed and ignored.
-
-   If the input cannot be parsed into the literal BNF literal,
-   the function returns `false` and sets the error number in @ref ec to the
-   appropriate error number defined in @ref boost::urls::error.
+   The initial character in the range is immediately consumed and ignored.
 
    @par Example
+
    @code
    std::string str = "Hello.cpp";
-   auto it = str.data();
-   auto const end = it + str.size();
+   char* it = str.data();
+   char* const end = it + str.size();
    if (bnf::parse(it, end, ec, 'H')) {
-       std::cout << "str begins with 'H'" << std::endl;
+       std::cout << "str begins with 'H'\n";
    }
    @endcode
 
    @par Exception Safety
-   Calls to other `parse` overloads may throw, although they should
-   prefer setting @ref ec instead.
 
-   @return `true` if the first char in the range matches @ref ch successfully,
-   `false` otherwise
+   Throws nothing
 
-   @param[in,out] it An iterator to the first element in the range.
-   At the end of this function, @ref it points to one past the last
+   @return `true` if the first char in the range matches `ch` successfully
+
+   @param it An iterator to the first element in the range.
+   At the end of this function, `it` points to one past the last
    element parsed in the range.
 
-   @param[in] end An iterator to one past the last element in the range
+   @param end An iterator to one past the last element in the range
+
+   @param ec Set to the error, if any occurred
 
-   @param[out] ec The error code this function should set @ref if it fails
+   @param ch A literal character
+
+   @see
+       @ref error.
```

Comment:
You don't need to ref these error because it is implied. Otherwise we end up with ref error on almost every function in the library, which is redundant.

Reply:
OK.